### PR TITLE
Role management POST endpoint

### PR DESCRIPTION
**Pull request by mgwalker**
Adds an API route for creating a new role.  The route is a `POST` to `/roles` and it expects a JSON object:

```json
{
  name: 'new role name',
  activities: [ 1, 2, 3 ]
}
```

...where the name must be unique and the values of the activities array must be numeric and correspond to IDs of activities known to the system. Will return a 400 if the body is invalid or 201 and the new role object if the creation is successful.

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- ~~The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented~~
- [x] The change has been documented
  - [x] Associated OpenAPI documentation has been updated

### This feature is done when...
- ~~Design has approved the experience~~
- ~~Product has approved the experience~~

### CODE REVIEW

**Comment by brendansudol** [comment]
these continual codeclimate false positives are getting quite annoying IMO

**Comment by mgwalker** [comment]
Yeah.  It'd be cool if we could configure it, but unfortunately...  Are there are other code-quality services we could use instead, that's either configurable or maybe more simply doesn't have these checks that go beyond what our linter does?

File: api/routes/roles/post.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      `Activity IDs ${JSON.stringify(invalidIDs).replace(
+        /\[(.+)\]/,
+        '$1'
+      )} are invalid`
+    );
+  }
+
+  return {
+    name: role.name,
+    activities: role.activities.filter(id => !Number.isNaN(Number(id)))
```
Comment (brendansudol) [inline api/routes/roles/post.js]: is this filter necessary after the validation check on #18 checking for numbers only?

**Comment by brendansudol** [comment]
i'm not aware of other/better ode-quality services, alas

File: api/routes/roles/post.js (hunk 1 above)
Comment (mgwalker) [reply api/routes/roles/post.js]: 😂 Nope.  How silly.  I'll remove in a future PR - I need to run `prettier` on the whole API codebase.  I've finally setup Atom to run it automatically on every save so hopefully it won't be a big deal going forward, but still need to clean up the past...  😬

Comment (brendansudol) [reply api/routes/roles/post.js]: nice - the atom-prettier on save hook is pretty 🤘

**Pull request merged**